engines/wasmer/example_test.go

```diff
@@ -15,9 +15,9 @@ func Example_custom() {
 	ctx := context.Background()
 
 	// Configure waPC to use a specific wasmer feature.
-	e := EngineWithRuntime(func() (*wasmer.Engine, error) {
+	e := EngineWith(WithRuntime(func() (*wasmer.Engine, error) {
```

Comment:
It's probably a good time to make the API across engines a little more idiomatic for Go. I'm thinking that each engine package should have either a `New` or `NewEngine` that excepts variadic options; e.g. `New(options ...Option)` that would deprecate `Engine()`. Not sure why this wasn't done initially but the good news is that it can be added this way without breaking compatibility.

WDYT @codefromthecrypt ? I think you mentioned something like this in one of your comments.

Reply:
works for me!

Reply:
@JanFalkin would you like to do the honors?

Reply:
Will do.  Got a little swamped today but before the end of the 4th, I'll have it pushed

Reply:
I forgot to mention.  I have pushed some more changes.  Is the implementation as expected?

Reply:
Drop the `With`... just `Engine` :)

```go
Engine(WithRuntime(func() (*wasmer.Engine, error) {
```

I also see that you changed the signature for `New` which breaks backward compatibility. I'd prefer to keep the changes focused on the new `Engine` func.

Reply:
Sorry forgot to post.  I put the New signature back.  Is this what we are looking for now?